v03_pipeline/lib/misc/pedigree.py

```diff
@@ -107,56 +124,34 @@ def parse_collateral_lineage(
         # A sample_i that is siblings with sample_j, will list sample_j as as sibling, but
         # sample_j will not list sample_i as a sibling.  Relationships only appear in the
         # ibd table a single time, so we only need to check the pairing once.
-        for sample_i, sample_j in itertools.combinations(samples.keys(), 2):
-            # If other sample is already related, continue
-            if sample_j in {
-                samples[sample_i].mother,
-                samples[sample_i].father,
-                samples[sample_i].maternal_grandmother,
-                samples[sample_i].maternal_grandfather,
-                samples[sample_i].paternal_grandmother,
-                samples[sample_i].paternal_grandfather,
-            }:
+        for sample_i, sample_j in itertools.combinations(samples.values(), 2):
+            # If sample is already related from direct relationships, continue
+            if sample_i.is_in_direct_lineage(sample_j):
                 continue
 
             # If both parents are identified and the same, samples are siblings.
             if (
-                samples[sample_i].mother
-                and samples[sample_i].father
-                and (samples[sample_i].mother == samples[sample_j].mother)
-                and (samples[sample_i].father == samples[sample_j].father)
+                sample_i.mother
+                and sample_i.father
+                and (sample_i.mother == sample_j.mother)
+                and (sample_i.father == sample_j.father)
             ):
-                samples[sample_i].siblings.append(
-                    sample_j,
-                )
+                sample_i.siblings.append(sample_j.sample_id)
                 continue
 
             # If only a single parent is identified and the same, samples are half siblings
-            if (
-                samples[sample_i].mother
-                and samples[sample_i].mother == samples[sample_j].mother
-            ) or (
-                samples[sample_i].father
-                and samples[sample_i].father == samples[sample_j].father
+            if (sample_i.mother and sample_i.mother == sample_j.mother) or (
+                sample_i.father and sample_i.father == sample_j.father
             ):
-                samples[sample_i].half_siblings.append(
-                    sample_j,
-                )
+                sample_i.half_siblings.append(sample_j.sample_id)
```

Comment:
remind me, I don't work with 'continue' much - full siblings aren't classified as half-siblings as well, right ?

Reply:
yep!  `continue` means to skip all subsequent steps... it's basically the same as an `else` (which might be clearer... `continue` is just a style choice).